Analytic Updates - 01262021

MHaggis opened the pull request:
New Analytic Stories:
- Trusted Developer Utilities Proxy Execution
- Trusted Developer Utilities Proxy Execution MSBuild

New Analytics:

Trusted Developer Utilities Proxy Execution MSBuild
- Suspicious MSBuild Rename (Passed)
- Suspicious MSBuild Spawn (Passed)
- Suspicious msbuild path (Passed)


Trusted Developer Utilities Proxy Execution
- Suspicious microsoft workflow compiler rename (Passed)
- Suspicious microsoft workflow compiler usage (Passed)

Suspicious MSHTA Activity
- Suspicious mshta child process (Passed)
- Detect MSHTA Url in Command Line (Passed)
- Detect Rundll32 Inline HTA Execution (Passed)
- Suspicious mshta spawn (Passed)
- Detect mshta inline hta execution (Passed)

Ryuk Ransomware 
- NLTest Domain Trust Discovery (Passed)
- WBAdmin Delete System Backups (Passed)

Commit:
T1127.001 MSbuild

MSbuild detections and test.

Commit:
Revert "T1127.001 MSbuild"

This reverts commit 0eba4877ffc44275e518a47832513ad145da250e.

Commit:
Revert "Revert "T1127.001 MSbuild""

This reverts commit e27ad1399a492252c972257e11be9262818f62cc.

Commit:
Revert "Revert "Revert "T1127.001 MSbuild"""

This reverts commit 9617d4d9b3db0f7ec0a43cae3da298a42da9ceb5.

Commit:
Revert "Revert "Revert "Revert "T1127.001 MSbuild""""

This reverts commit fd5e4adfb187b87ad878d1ded83c1ff344ccf033.

Commit:
MSBuild

fixes

Commit:
ID Collision

Commit:
Update suspicious_msbuild_netconn.yml

Commit:
MSHTA

Modified inlne mshta execution with the `about` protocol handler.
Added new mshta.exe child process analytic

Commit:
MSBuild Story

Commit:
Update trusted_developer_utilities_proxy_execution_msbuild.yml

Commit:
Msbuild Story name and analytics fix

github-actions[bot] committed:
Branch was auto-updated.

github-actions[bot] committed:
Branch was auto-updated.

github-actions[bot] committed:
Branch was auto-updated.

github-actions[bot] committed:
Branch was auto-updated.

github-actions[bot] committed:
Branch was auto-updated.

P4T12ICK commented:
This PR was created by Automated Detection Testing Service :robot:

Tested detections:

Detect renamed msbuild execution :x:
- diskUsage: 270336
- runDuration: 0.7020000000000001s
- scanCount: 12

P4T12ICK commented:
This PR was created by Automated Detection Testing Service :robot:

Tested detections:

Detect moved msbuild execution :x:
- diskUsage: 339968
- runDuration: 0.937s
- scanCount: 0

P4T12ICK commented:
This PR was created by Automated Detection Testing Service :robot:

Tested detections:

Detect wmiprvse spawn msbuild :x:
- diskUsage: 339968
- runDuration: 0.921s
- scanCount: 0

Commit:
Timestamp

Trying to fix :)

P4T12ICK commented:
This PR was created by Automated Detection Testing Service :robot:

Tested detections:

Detect moved msbuild execution :x:
- diskUsage: 335872
- runDuration: 0.873s
- scanCount: 0

P4T12ICK commented:
This PR was created by Automated Detection Testing Service :robot:

Tested detections:

Detect wmiprvse spawn msbuild :x:
- diskUsage: 344064
- runDuration: 0.961s
- scanCount: 0

P4T12ICK commented:
This PR was created by Automated Detection Testing Service :robot:

Tested detections:

Detect renamed msbuild execution :x:
- diskUsage: 270336
- runDuration: 0.706s
- scanCount: 13

github-actions[bot] committed:
Branch was auto-updated.

Commit:
Removing netconn

P4T12ICK commented:
This PR was created by Automated Detection Testing Service :robot:

Tested detections:

Detect renamed msbuild execution :x:
- diskUsage: 270336
- runDuration: 0.6960000000000001s
- scanCount: 10

P4T12ICK commented:
This PR was created by Automated Detection Testing Service :robot:

Tested detections:

Detect wmiprvse spawn msbuild :x:
- diskUsage: 335872
- runDuration: 0.9550000000000001s
- scanCount: 0

Commit:
Update suspicious_msbuild_rename.yml

Commit:
Update suspicious_msbuild_rename.yml

Commit:
URL Fixes

Fixing all the URLs

Commit:
Added detection testing service results inDetect wmiprvse spawn msbuild

P4T12ICK commented:
This PR was created by Automated Detection Testing Service :robot:

Tested detections:

Detect wmiprvse spawn msbuild :white_check_mark:
- diskUsage: 364544
- runDuration: 2.987s
- scanCount: 3192

Commit:
Added detection testing service results inDetect moved msbuild execution

P4T12ICK commented:
This PR was created by Automated Detection Testing Service :robot:

Tested detections:

Detect moved msbuild execution :white_check_mark:
- diskUsage: 368640
- runDuration: 2.922s
- scanCount: 3192

Commit:
Added detection testing service results inDetect renamed msbuild execution

Commit:
Merge branch 'msbuild' of https://github.com/splunk/security-content into msbuild

P4T12ICK commented:
This PR was created by Automated Detection Testing Service :robot:

Tested detections:

Detect renamed msbuild execution :white_check_mark:
- diskUsage: 294912
- runDuration: 2.599s
- scanCount: 3208

Commit:
Microsoft Workflow Compiler

Adding analytics for microsoft.workflow.compiler.exe.

Commit:
Merge branch 'msbuild' of https://github.com/splunk/security-content into msbuild

Commit:
Update trusted_developer_utilities_proxy_execution.yml

Commit:
Added detection testing service results inDetect microsoft.workflow.compiler.exe rename

P4T12ICK commented:
This PR was created by Automated Detection Testing Service :robot:

Tested detections:

Detect microsoft.workflow.compiler.exe rename :white_check_mark:
- diskUsage: 299008
- runDuration: 8.677s
- scanCount: 14230

Commit:
Added detection testing service results inDetect microsoft.workflow.compiler.exe usage

Commit:
Merge branch 'msbuild' of https://github.com/splunk/security-content into msbuild

P4T12ICK commented:
This PR was created by Automated Detection Testing Service :robot:

Tested detections:

Detect microsoft.workflow.compiler.exe usage :white_check_mark:
- diskUsage: 360448
- runDuration: 9.65s
- scanCount: 14217

MHaggis committed:
Merge branch 'develop' into msbuild

Commit:
rundll32 inline hta

Rundll32 inline hta execution

Commit:
Added detection testing service results inDetect rundll32 executing hta with inline protocols

P4T12ICK commented:
This PR was created by Automated Detection Testing Service :robot:

Tested detections:

Detect rundll32 executing hta with inline protocols :white_check_mark:
- diskUsage: 360448
- runDuration: 6.6290000000000004s
- scanCount: 9131

Commit:
Update detect_rundll32_inline_hta_execution.yml

Commit:
MSHTA

Updated story
added mshta url in cmdline analytic

Commit:
Added detection testing service results inDetect mshta url in command line

P4T12ICK commented:
This PR was created by Automated Detection Testing Service :robot:

Tested detections:

Detect mshta url in command line :white_check_mark:
- diskUsage: 348160
- runDuration: 6.5360000000000005s
- scanCount: 9131

Commit:
rename mshta inline hta execution

Commit:
Merge branch 'msbuild' of https://github.com/splunk/security-content into msbuild

Commit:
Update detect_mshta_inline_hta_execution.yml

Commit:
Update detect_mshta_inline_hta_execution.yml

Commit:
Update detect_mshta_inline_hta_execution.yml

Commit:
mshta renamed

Commit:
Update detect_mshta_renamed.yml

MHaggis committed:
Merge branch 'develop' into msbuild

Commit:
Suspicious mshta spawn

Suspicious mshta spawn

Commit:
ID changes

ID changes and a test file needed moved

Commit:
renamed testing

Commit:
Added detection testing service results inDetect mshta spawn

P4T12ICK commented:
This PR was created by Automated Detection Testing Service :robot:

Tested detections:

Detect mshta spawn :white_check_mark:
- diskUsage: 360448
- runDuration: 6.605s
- scanCount: 9131

Commit:
MSBuild story update

github-actions[bot] committed:
Branch was auto-updated.

Commit:
Update trusted_developer_utilities_proxy_execution_msbuild.yml

Commit:
Merge branch 'msbuild' of https://github.com/splunk/security-content into msbuild

Commit:
Update trusted_developer_utilities_proxy_execution_msbuild.yml

Commit:
Update trusted_developer_utilities_proxy_execution_msbuild.yml

Commit:
Update trusted_developer_utilities_proxy_execution_msbuild.yml

Commit:
Update trusted_developer_utilities_proxy_execution_msbuild.yml

Commit:
Update trusted_developer_utilities_proxy_execution_msbuild.yml

Commit:
Update trusted_developer_utilities_proxy_execution_msbuild.yml

Commit:
Update trusted_developer_utilities_proxy_execution_msbuild.yml

Commit:
Update trusted_developer_utilities_proxy_execution_msbuild.yml

Commit:
Update trusted_developer_utilities_proxy_execution_msbuild.yml

Commit:
Update trusted_developer_utilities_proxy_execution_msbuild.yml

Commit:
Update suspicious_mshta_activity.yml

github-actions[bot] committed:
Branch was auto-updated.

github-actions[bot] committed:
Branch was auto-updated.

Commit:
Adding WBAdmin - delete system backups

Deleting system backups with WBAdmin

Commit:
Added detection testing service results inWBAdmin Delete System Backups

P4T12ICK commented:
This PR was created by Automated Detection Testing Service :robot:

Tested detections:

WBAdmin Delete System Backups :white_check_mark:
- diskUsage: 348160
- runDuration: 1.222s
- scanCount: 29

github-actions[bot] committed:
Branch was auto-updated.

Commit:
T1482 - NLTest usage

NLTest execution to discover domain trusts.

P4T12ICK commented:
This PR was created by Automated Detection Testing Service :robot:

Tested detections:

NLTest Domain Trust Discovery :x:
- diskUsage: 126976
- runDuration: 0.004s
- scanCount: 0

Commit:
Merge branch 'develop' into msbuild

P4T12ICK commented:
This PR was created by Automated Detection Testing Service :robot:

Tested detections:

NLTest Domain Trust Discovery :x:
- diskUsage: 126976
- runDuration: 0.003s
- scanCount: 0

Commit:
minor updates

MHaggis committed:
Merge branch 'develop' into msbuild

Commit:
Removed mshta rename

github-actions[bot] committed:
Branch was auto-updated.

P4T12ICK commented:
This PR was created by Automated Detection Testing Service :robot:

Tested detections:

Detect mshta spawning a process :x:
- diskUsage: 376832
- runDuration: 0.9390000000000001s
- scanCount: 0

P4T12ICK commented:
This PR was created by Automated Detection Testing Service :robot:

Tested detections:

Detect mshta spawning a process :x:
- diskUsage: 368640
- runDuration: 0.917s
- scanCount: 0

Commit:
Added detection testing service results inNLTest Domain Trust Discovery

P4T12ICK commented:
This PR was created by Automated Detection Testing Service :robot:

Tested detections:

NLTest Domain Trust Discovery :white_check_mark:
- diskUsage: 430080
- runDuration: 1.1500000000000001s
- scanCount: 4

Commit:
Update suspicious_mshta_child_process.test.yml

Commit:
Merge branch 'msbuild' of https://github.com/splunk/security_content into msbuild

P4T12ICK commented:
This PR was created by Automated Detection Testing Service :robot:

Tested detections:

Detect mshta spawning a process :x:
- diskUsage: 376832
- runDuration: 0.937s
- scanCount: 0

Commit:
Update suspicious_mshta_child_process.yml

Commit:
wrong url

URL was wrong :(

Commit:
Added detection testing service results inDetect mshta spawning a process

P4T12ICK commented:
This PR was created by Automated Detection Testing Service :robot:

Tested detections:

Detect mshta spawning a process :white_check_mark:
- diskUsage: 397312
- runDuration: 11.790000000000001s
- scanCount: 17702

github-actions[bot] committed:
Branch was auto-updated.

github-actions[bot] committed:
Branch was auto-updated.

github-actions[bot] committed:
Branch was auto-updated.

P4T12ICK commented on detections/endpoint/suspicious_msbuild_path.yml:
```diff
@@ -0,0 +1,43 @@
+search: '| tstats `security_content_summariesonly` count values(Processes.process_name)
+  as process_name values(Processes.process) as process min(_time) as firstTime max(_time)
+  as lastTime from datamodel=Endpoint.Processes where Processes.process_name=msbuild.exe
+  AND (Processes.process_path!=c:\\windows\\microsoft.net\\framework*\\v4.0.30319\\*)
```
Can we make that independent from the version?

MHaggis replied:
Removed the specified version and updated false positive and description to reflect Visual Studio usage of `msbuild.exe`.

P4T12ICK commented on detections/endpoint/suspicious_msbuild_path.yml:
```diff
@@ -0,0 +1,43 @@
+  analytics_story:
+  - Trusted Developer Utilities Proxy Execution MSBuild
+  mitre_attack_id:
+  - T1127.001, T1036.003
```
A array in yml format is:
mitre_attack_id:
- T1127.001
- T1036.003

P4T12ICK commented on detections/endpoint/suspicious_msbuild_rename.yml:
```diff
@@ -0,0 +1,42 @@
+  analytics_story:
+  - Trusted Developer Utilities Proxy Execution MSBuild
+  mitre_attack_id:
+  - T1127.001, T1036.003
```
same as above

P4T12ICK commented on detections/endpoint/suspicious_msbuild_spawn.yml:
```diff
@@ -0,0 +1,44 @@
+search: '| tstats `security_content_summariesonly` count values(Processes.process_name)
+  as process_name values(Processes.process) as process min(_time) as firstTime max(_time)
+  as lastTime from datamodel=Endpoint.Processes where Processes.parent_process_name=wmiprvse.exe
+  AND Processes.process_name=msbuild.exe by Processes.dest Processes.parent_process
```
Only FYI: AND are not really needed in Splunk. A space is the same as an AND

github-actions[bot] committed:
Branch was auto-updated.

Commit:
Added detection testing service results inDetect renamed msbuild execution

Commit:
Merge branch 'msbuild' of https://github.com/splunk/security_content into msbuild

P4T12ICK commented:
This PR was created by Automated Detection Testing Service :robot:

Tested detections:

Detect renamed msbuild execution :white_check_mark:
- diskUsage: 299008
- runDuration: 2.657s
- scanCount: 3205

github-actions[bot] committed:
Branch was auto-updated.

github-actions[bot] committed:
Branch was auto-updated.

github-actions[bot] committed:
Branch was auto-updated.

github-actions[bot] committed:
Branch was auto-updated.

github-actions[bot] committed:
Branch was auto-updated.

Commit:
Resolved yml formatting

github-actions[bot] committed:
Branch was auto-updated.

github-actions[bot] committed:
Branch was auto-updated.

github-actions[bot] committed:
Branch was auto-updated.

github-actions[bot] committed:
Branch was auto-updated.

P4T12ICK reviewed (dismissed):
Looks good to me

github-actions[bot] committed:
Branch was auto-updated.

Commit:
Update suspicious_msbuild_path.yml

Modified to exclude all version path. Updated false positive info and description to reflect Visual Studio usage of msbuild.
